staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go

```diff
@@ -2366,3 +2368,48 @@ func TestNumberConversion(t *testing.T) {
 		}
 	}
 }
+
+var replaceRawExtensionPatchTestCases = []StrategicMergePatchRawTestCase{
+	{
+		Description: "replace RawExtension field",
+		StrategicMergePatchRawTestCaseData: StrategicMergePatchRawTestCaseData{
+			Original: []byte(`
+replacingItem:
+  Some: Generic
+  Yaml: Inside
+  The: RawExtension
+  Field: Period
+other: current-other
+`),
+			Modified: []byte(`
+replacingItem:
+  Newly: Modified
+  Yaml: Inside
+  The: RawExtension
+other: current-other
+`),
+			TwoWay: []byte(`
+replacingItem:
+  Newly: Modified
+  Yaml: Inside
+  The: RawExtension
+`),
+		},
+	},
+}
+
+func TestReplaceWithRawExtension(t *testing.T) {
+	for _, c := range replaceRawExtensionPatchTestCases {
+		original, twoWay, modified, _ := twoWayRawTestCaseToJSONOrFail(t, c)
+		actualPatch, err := CreateTwoWayMergePatch(original, modified, mergeItem)
```

Comment:
also test applying the patch, making sure it applies successfully, and the result is what you expect (for both two and three-way)